src/main/java/com/minecolonies/coremod/entity/ai/citizen/enchanter/EntityAIWorkEnchanter.java

```diff
@@ -408,4 +418,52 @@ private void resetDraining()
     {
         return BuildingEnchanter.class;
     }
+
+    private static class RecordingItemHandler implements IItemHandler
```

Comment:
This might actually be usefull in other locations.
Could you put this in the API as a proper class?

Also: There is IModifyableItemHandler, which has a direct set method, as such you might rather want to make sure that is covered also somehow.